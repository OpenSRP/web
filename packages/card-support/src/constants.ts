// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
