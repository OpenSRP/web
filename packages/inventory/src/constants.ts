// opensrp api strings
export const OPENSRP_API_BASE_URL = 'https://mg-eusm-staging.smartregister.org/opensrp/rest/';

// router routes
export const INVENTORY_SERVICE_POINT_LIST_VIEW = '/inventory/list';
export const INVENTORY_SERVICE_POINT_PROFILE_VIEW = '/inventory/list';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'inventory';

// magic strings
export const SEARCH_QUERY_PARAM = 'querySearch';
export const TABLE_PAGE_SIZE = 5;
export const TABLE_PAGE_SIZE_OPTIONS = ['5', '10', '20', '50', '100'];

export const tablePaginationOptions = {
  showQuickJumper: true,
  showSizeChanger: true,
  defaultPageSize: TABLE_PAGE_SIZE,
  pageSizeOptions: TABLE_PAGE_SIZE_OPTIONS,
};

//endpoints
export const GET_INVENTORY_BY_SERVICE_POINT = 'stockresource/servicePointId/';

//payload
export const inventory1 = {
  stockId: '5dfcaecd-430a-4284-972f-9acf87775a9b1',
  productName: 'Scale1',
  unicefSection: 'Health1',
  quantity: 4,
  deliveryDate: '2020-02-01',
  accountabilityEndDate: '2020-10-10',
  donor: 'ADB',
  servicePointId: '903971',
  poNumber: 1111,
  serialNumber: '1234serial2',
  providerId: 'providerId',
};

export const inventory2 = {
  stockId: '5dfcaecd-430a-4284-972f-9acf87775a9b2',
  productName: 'Scale2',
  unicefSection: 'Health2',
  quantity: 4,
  deliveryDate: '2020-02-01',
  accountabilityEndDate: '2020-10-10',
  donor: 'ADB',
  servicePointId: '903972',
  poNumber: 1112,
  serialNumber: '1234serial2',
  providerId: 'providerId',
};
