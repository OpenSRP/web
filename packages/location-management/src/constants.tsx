// URLs
export const URL_ALL_LOCATION_TAGS = '/location-tag';
export const URL_DELETE_LOCATION_TAGS = '/location-tag/delete';
export const KEYCLOAK_API_BASE_URL ='https://opensrp-stage.smartregister.org/opensrp/rest';
