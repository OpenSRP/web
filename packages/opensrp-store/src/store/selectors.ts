import { reducerName as sessionReducerName } from '@onaio/session-reducer';
import { Store } from 'redux';
import { Dictionary } from '@onaio/utils';
import { createSelector } from 'reselect';

/** get API Token from the Redux store
 *
 * @param {any} state - the redux store
 */
interface StoreState {
  session: {
    [key: string]: Dictionary;
  };
}

interface APIFilters {
  accessToken?: boolean;
  apiToken?: boolean;
  providerState?: boolean;
}

/** Fetch access token from store if access_token filter val is true
 *
 * @param _ - redux store
 * @param props - api filters object
 * @returns {boolean}
 */

export const fetchAccessToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.accessToken || null;

/** Fetch api token from store if api_token filter val is true
 *
 * @param _ redux store
 * @param props api filters object
 * @returns {boolean}
 */

export const fetchApiToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.apiToken || null;

/** Fetch oauth provider state from store if providerState filter val is true
 *
 * @param _ redux store
 * @param props api filters object
 * @returns {boolean}
 */

export const fetchOauthProviderState = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.providerState || null;

/** Gets extra data object from store
 *
 * @param state - redux store
 */

export const getExtraData = (state: Partial<Store>): Dictionary =>
  (state as StoreState)[sessionReducerName].extraData;

/**
 * @param {any} state - the redux store
 * @returns {Object} - the states
 */

export const makeAPIStateSelector = () =>
  createSelector(
    [fetchAccessToken, fetchApiToken, fetchOauthProviderState, getExtraData],
    (accessToken, apiToken, oauthState, extraData) => {
      if (accessToken) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.access_token) || null;
      } else if (apiToken) {
        return extraData.api_token || null;
      } else if (oauthState) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.state) || null;
      }
    }
  );
