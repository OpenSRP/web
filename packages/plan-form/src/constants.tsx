export const A1 = 'A1';
export const A2 = 'A2';
export const B1 = 'B1';
export const B2 = 'B2';
export const LOW_PRIORITY = 'low-priority';
export const MEDIUM_PRIORITY = 'medium-priority';
export const HIGH_PRIORITIY = 'high-priority';
export const TRUE = 'True';
export const FALSE = 'False';
export const DISABLED = 'Disabled';
export const INTERNAL = 'internal';
export const IGNORE = 'ignore';
export const INTERVENTION_TYPE_CODE = 'interventionType';
export const FI_STATUS_CODE = 'fiStatus';
export const FI_REASON_CODE = 'fiReason';
export const OPENSRP_EVENT_ID_CODE = 'opensrpEventId';
export const CASE_NUMBER_CODE = 'caseNum';
export const TASK_GENERATION_STATUS_CODE = 'taskGenerationStatus';
export const TEAM_ASSIGNMENT_STATUS_CODE = 'teamAssignmentStatus';
export const CASE_TRIGGERED = 'Case Triggered';
export const ROUTINE = 'Routine';
export const INVESTIGATION = 'Investigation';
export const DAYS = 'days';
export const CONDITION = 'condition';
export const TRIGGER = 'trigger';

// plan activity codes
export const CASE_CONFIRMATION_ACTIVITY_CODE = 'caseConfirmation';
export const FAMILY_REGISTRATION_ACTIVITY_CODE = 'familyRegistration';
// opensrp api strings
export const PLAN_LIST_URL = '/plans/list';
export const OPENSRP_API_BASE_URL = '';
export const OPENSRP_PLAN_ENDPOINT = 'plans';