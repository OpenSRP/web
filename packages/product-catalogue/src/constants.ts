// opensrp api strings
export const OPENSRP_PRODUCT_CATALOGUE = 'rest/product-catalogue';

// router routes
export const CATALOGUE_LIST_VIEW_URL = '/product-catalogue';
export const CATALOGUE_EDIT_VIEW_URL = '/product-catalogue/edit';
export const CATALOGUE_CREATE_VIEW_URL = 'product-catalogue/new';
export const HOME_URL = '/';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'product-catalogue';

/** route params for product-catalogue pages */
export interface RouteParams {
  productId?: string;
}
