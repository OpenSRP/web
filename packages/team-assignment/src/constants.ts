export const OPENSRP_API_BASE_URL = 'https://opensrp-staging.smartregister.org/opensrp/rest/';

// router routes
export const TEAM_ASSIGNMENT_LIST_VIEW_URL = '/team-assignment';
export const TEAM_ASSIGNMENT_EDIT_VIEW_URL = '/team-assignment/edit';
export const TEAM_ASSIGNMENT_CREATE_VIEW_URL = 'team-assignment/new';
export const HOME_URL = '/';
export const LOCATION_HIERARCHY_ENDPOINT = 'location/hierarchy';
export const PLANS_ENDPOINT = 'plans';
export const ASSIGNMENTS_ENDPOINT = 'organization/assignLocationsAndPlans';
export const SUCCESSFULLY_ASSIGNED_TEAMS = 'Successfully Assigned Teams';
export const ENTER_TEAM_NAME = 'Enter a Team name';

// other constants
/** namespace for the keys attached to the columns */
export const TableColumnsNamespace = 'team-assignment';

// Table view strings

export const TEAM_ASSIGNMENT_PAGE_TITLE = 'Team Assignment';
export const SAVE = 'Save';
export const CANCEL = 'Cancel';
