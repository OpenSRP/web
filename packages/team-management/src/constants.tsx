export const API_BASE_URL = process.env.REACT_APP_OPENSRP_API_BASE_URL;

//url
export const URL_ADD_TEAM = 'teams/add';
export const URL_EDIT_TEAM = 'teams/edit/';

//
// Teams detail
//
export const TEAMS_GET = 'organization';

export const PRACTITIONER_GET = 'practitioner';
export const PRACTITIONER_POST = 'practitionerRole/add';
export const TEAM_PRACTITIONERS = 'organization/practitioner/';
export const PRACTITIONER_ROLE_ADD = 'practitionerRole/add';

export const TEAM_UUID = 'team-uuid';

export const ERROR_OCCURRED = 'An error occurred';
