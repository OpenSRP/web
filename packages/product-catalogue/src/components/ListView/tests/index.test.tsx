import React from 'react';
import { ConnectedProductCatalogueList } from '..';
import { store } from '@opensrp/store';
import { createBrowserHistory } from 'history';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { product1 } from '../../../ducks/productCatalogue/tests/fixtures';
import { CATALOGUE_LIST_VIEW_URL } from '../../../constants';
import { Helmet } from 'react-helmet';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

describe('CreateEditProduct Page', () => {
  afterEach(() => {
    fetch.resetMocks();
  });

  it('renders correctly', async () => {
    fetch.mockResponse(JSON.stringify([]));
    const props = {
      history,
      location: {
        hash: '',
        pathname: `${CATALOGUE_LIST_VIEW_URL}`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: {},
        path: `${CATALOGUE_LIST_VIEW_URL}`,
        url: `${CATALOGUE_LIST_VIEW_URL}`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedProductCatalogueList {...props}></ConnectedProductCatalogueList>
        </Router>
      </Provider>
    );

    /** loading view */
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"Loading...Fetching product CataloguePlease wait, as we fetch the product Catalogue."`
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    expect(wrapper.text()).toMatchInlineSnapshot(
      `"Product Catalogue (0) + Product CatalogueProduct NameIDActionsNo Data"`
    );

    // find ant table
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`table rows ${index}`);
    });

    // check if page title is correct
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual('Product Catalogue (0)');

    // details view is not displayed
    expect(wrapper.find('.view-details-content')).toHaveLength(0);
  });

  it('renders correctly with detailView', async () => {
    fetch.mockResponse(JSON.stringify([product1]));
    const props = {
      history,
      location: {
        hash: '',
        pathname: `${CATALOGUE_LIST_VIEW_URL}/1`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: { productId: '1' },
        path: `${CATALOGUE_LIST_VIEW_URL}/:productId`,
        url: `${CATALOGUE_LIST_VIEW_URL}/1`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedProductCatalogueList {...props}></ConnectedProductCatalogueList>
        </Router>
      </Provider>
    );

    /** loading view */
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"Loading...Fetching product CataloguePlease wait, as we fetch the product Catalogue."`
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    // details view is not displayed
    expect(wrapper.find('.view-details-content')).toHaveLength(0);

    expect(toJson(wrapper.find('view-details-content'))).toMatchSnapshot('View details');
  });

  it('shows broken page', async () => {
    fetch.mockReject(new Error('Something went wrong'));
    const props = {
      history,
      location: {
        hash: '',
        pathname: `${CATALOGUE_LIST_VIEW_URL}/6`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: { productId: '1' },
        path: `${CATALOGUE_LIST_VIEW_URL}/:productId`,
        url: `${CATALOGUE_LIST_VIEW_URL}/6`,
      },
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedProductCatalogueList {...props}></ConnectedProductCatalogueList>
        </Router>
      </Provider>
    );

    /** loading view */
    expect(wrapper.text()).toMatchInlineSnapshot(
      `"Loading...Fetching product CataloguePlease wait, as we fetch the product Catalogue."`
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
      wrapper.update();
    });

    /** error view */
    expect(wrapper.text()).toMatchInlineSnapshot(`"ErrorSomething went wrongGo BackBack Home"`);
  });
});
