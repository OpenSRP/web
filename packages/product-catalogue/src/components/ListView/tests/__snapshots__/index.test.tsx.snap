// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEditProduct Page missing product for detail view: View details 1`] = `null`;

exports[`CreateEditProduct Page renders correctly with detailView: View details 1`] = `null`;

exports[`CreateEditProduct Page renders correctly: table rows 0 1`] = `
<tr>
  <Cell
    additionalProps={
      Object {
        "className": "ant-table-column-has-sorters",
        "onClick": [Function],
      }
    }
    className="ant-table-column-sort"
    colEnd={0}
    colSpan={1}
    colStart={0}
    column={
      Object {
        "className": "ant-table-column-sort",
        "dataIndex": "productName",
        "defaultSortOrder": "descend",
        "key": "product-catalogue-productName",
        "onHeaderCell": [Function],
        "sorter": [Function],
        "title": <ForwardRef
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="top"
          title="Click to cancel sort"
          transitionName="zoom-big-fast"
        >
          <div
            className="ant-table-column-sorters-with-tooltip"
          >
            <div
              className="ant-table-column-sorters"
            >
              <span>
                Product Name
              </span>
              <span
                className="ant-table-column-sorter ant-table-column-sorter-full"
              >
                <span
                  className="ant-table-column-sorter-inner"
                >
                  <ForwardRef(CaretUpOutlined)
                    className="ant-table-column-sorter-up"
                  />
                  <ForwardRef(CaretDownOutlined)
                    className="ant-table-column-sorter-down active"
                  />
                </span>
              </span>
            </div>
          </div>
        </ForwardRef>,
      }
    }
    component="th"
    firstFixLeft={false}
    firstFixRight={false}
    key="product-catalogue-productName"
    lastFixLeft={false}
    lastFixRight={false}
    prefixCls="ant-table"
    rowSpan={1}
    rowType="header"
  >
    <th
      className="ant-table-cell ant-table-column-sort ant-table-column-has-sorters"
      colSpan={null}
      onClick={[Function]}
      rowSpan={null}
      style={Object {}}
    >
      <Tooltip
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement="top"
        title="Click to cancel sort"
        transitionName="zoom-big-fast"
      >
        <ForwardRef(Tooltip)
          arrowContent={
            <span
              className="ant-tooltip-arrow-content"
            />
          }
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          builtinPlacements={
            Object {
              "bottom": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tc",
                  "bc",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "bottomLeft": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tl",
                  "bl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "bottomRight": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tr",
                  "br",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "left": Object {
                "ignoreShake": true,
                "offset": Array [
                  -4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "cr",
                  "cl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "leftBottom": Object {
                "ignoreShake": true,
                "offset": Array [
                  -4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "br",
                  "bl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "leftTop": Object {
                "ignoreShake": true,
                "offset": Array [
                  -4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tr",
                  "tl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "right": Object {
                "ignoreShake": true,
                "offset": Array [
                  4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "cl",
                  "cr",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "rightBottom": Object {
                "ignoreShake": true,
                "offset": Array [
                  4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "bl",
                  "br",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "rightTop": Object {
                "ignoreShake": true,
                "offset": Array [
                  4,
                  0,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tl",
                  "tr",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "top": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  -4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "bc",
                  "tc",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "topLeft": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  -4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "bl",
                  "tl",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
              "topRight": Object {
                "ignoreShake": true,
                "offset": Array [
                  0,
                  -4,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "br",
                  "tr",
                ],
                "targetOffset": Array [
                  0,
                  0,
                ],
              },
            }
          }
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          onPopupAlign={[Function]}
          onVisibleChange={[Function]}
          overlay="Click to cancel sort"
          overlayClassName=""
          placement="top"
          prefixCls="ant-tooltip"
          title="Click to cancel sort"
          transitionName="zoom-big-fast"
          visible={false}
        >
          <Trigger
            action={
              Array [
                "hover",
              ]
            }
            afterPopupVisibleChange={[Function]}
            arrowContent={
              <span
                className="ant-tooltip-arrow-content"
              />
            }
            arrowPointAtCenter={false}
            autoAdjustOverflow={true}
            autoDestroy={false}
            blurDelay={0.15}
            builtinPlacements={
              Object {
                "bottom": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tc",
                    "bc",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "bottomLeft": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tl",
                    "bl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "bottomRight": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tr",
                    "br",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "left": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    -4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "cr",
                    "cl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "leftBottom": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    -4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "br",
                    "bl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "leftTop": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    -4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tr",
                    "tl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "right": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "cl",
                    "cr",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "rightBottom": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "bl",
                    "br",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "rightTop": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    4,
                    0,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "tl",
                    "tr",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "top": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    -4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "bc",
                    "tc",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "topLeft": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    -4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "bl",
                    "tl",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
                "topRight": Object {
                  "ignoreShake": true,
                  "offset": Array [
                    0,
                    -4,
                  ],
                  "overflow": Object {
                    "adjustX": 1,
                    "adjustY": 1,
                  },
                  "points": Array [
                    "br",
                    "tr",
                  ],
                  "targetOffset": Array [
                    0,
                    0,
                  ],
                },
              }
            }
            defaultPopupVisible={false}
            destroyPopupOnHide={false}
            focusDelay={0}
            getDocument={[Function]}
            getPopupClassNameFromAlign={[Function]}
            hideAction={Array []}
            mask={false}
            maskClosable={true}
            mouseEnterDelay={0.1}
            mouseLeaveDelay={0.1}
            onPopupAlign={[Function]}
            onPopupVisibleChange={[Function]}
            overlay="Click to cancel sort"
            popup={[Function]}
            popupAlign={Object {}}
            popupClassName=""
            popupPlacement="top"
            popupStyle={Object {}}
            popupTransitionName="zoom-big-fast"
            popupVisible={false}
            prefixCls="ant-tooltip"
            showAction={Array []}
            title="Click to cancel sort"
            visible={false}
          >
            <div
              className="ant-table-column-sorters-with-tooltip"
              key="trigger"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="ant-table-column-sorters"
              >
                <span>
                  Product Name
                </span>
                <span
                  className="ant-table-column-sorter ant-table-column-sorter-full"
                >
                  <span
                    className="ant-table-column-sorter-inner"
                  >
                    <ForwardRef(CaretUpOutlined)
                      className="ant-table-column-sorter-up"
                    >
                      <AntdIcon
                        className="ant-table-column-sorter-up"
                        icon={
                          Object {
                            "icon": Object {
                              "attrs": Object {
                                "focusable": "false",
                                "viewBox": "0 0 1024 1024",
                              },
                              "children": Array [
                                Object {
                                  "attrs": Object {
                                    "d": "M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z",
                                  },
                                  "tag": "path",
                                },
                              ],
                              "tag": "svg",
                            },
                            "name": "caret-up",
                            "theme": "outlined",
                          }
                        }
                      >
                        <span
                          aria-label="caret-up"
                          className="anticon anticon-caret-up ant-table-column-sorter-up"
                          role="img"
                        >
                          <IconReact
                            className=""
                            icon={
                              Object {
                                "icon": Object {
                                  "attrs": Object {
                                    "focusable": "false",
                                    "viewBox": "0 0 1024 1024",
                                  },
                                  "children": Array [
                                    Object {
                                      "attrs": Object {
                                        "d": "M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z",
                                      },
                                      "tag": "path",
                                    },
                                  ],
                                  "tag": "svg",
                                },
                                "name": "caret-up",
                                "theme": "outlined",
                              }
                            }
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="caret-up"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              key="svg-caret-up"
                              viewBox="0 0 1024 1024"
                              width="1em"
                            >
                              <path
                                d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                key="svg-caret-up-svg-0"
                              />
                            </svg>
                          </IconReact>
                        </span>
                      </AntdIcon>
                    </ForwardRef(CaretUpOutlined)>
                    <ForwardRef(CaretDownOutlined)
                      className="ant-table-column-sorter-down active"
                    >
                      <AntdIcon
                        className="ant-table-column-sorter-down active"
                        icon={
                          Object {
                            "icon": Object {
                              "attrs": Object {
                                "focusable": "false",
                                "viewBox": "0 0 1024 1024",
                              },
                              "children": Array [
                                Object {
                                  "attrs": Object {
                                    "d": "M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z",
                                  },
                                  "tag": "path",
                                },
                              ],
                              "tag": "svg",
                            },
                            "name": "caret-down",
                            "theme": "outlined",
                          }
                        }
                      >
                        <span
                          aria-label="caret-down"
                          className="anticon anticon-caret-down ant-table-column-sorter-down active"
                          role="img"
                        >
                          <IconReact
                            className=""
                            icon={
                              Object {
                                "icon": Object {
                                  "attrs": Object {
                                    "focusable": "false",
                                    "viewBox": "0 0 1024 1024",
                                  },
                                  "children": Array [
                                    Object {
                                      "attrs": Object {
                                        "d": "M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z",
                                      },
                                      "tag": "path",
                                    },
                                  ],
                                  "tag": "svg",
                                },
                                "name": "caret-down",
                                "theme": "outlined",
                              }
                            }
                          >
                            <svg
                              aria-hidden="true"
                              className=""
                              data-icon="caret-down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              key="svg-caret-down"
                              viewBox="0 0 1024 1024"
                              width="1em"
                            >
                              <path
                                d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                key="svg-caret-down-svg-0"
                              />
                            </svg>
                          </IconReact>
                        </span>
                      </AntdIcon>
                    </ForwardRef(CaretDownOutlined)>
                  </span>
                </span>
              </div>
            </div>
          </Trigger>
        </ForwardRef(Tooltip)>
      </Tooltip>
    </th>
  </Cell>
  <Cell
    className=""
    colEnd={1}
    colSpan={1}
    colStart={1}
    column={
      Object {
        "dataIndex": "uniqueId",
        "key": "product-catalogue-uniqueId",
        "title": "ID",
      }
    }
    component="th"
    firstFixLeft={false}
    firstFixRight={false}
    key="product-catalogue-uniqueId"
    lastFixLeft={false}
    lastFixRight={false}
    prefixCls="ant-table"
    rowSpan={1}
    rowType="header"
  >
    <th
      className="ant-table-cell"
      colSpan={null}
      rowSpan={null}
      style={Object {}}
    >
      ID
    </th>
  </Cell>
  <Cell
    className=""
    colEnd={2}
    colSpan={1}
    colStart={2}
    column={
      Object {
        "key": "product-catalogue-actions",
        "render": [Function],
        "title": "Actions",
      }
    }
    component="th"
    firstFixLeft={false}
    firstFixRight={false}
    key="product-catalogue-actions"
    lastFixLeft={false}
    lastFixRight={false}
    prefixCls="ant-table"
    rowSpan={1}
    rowType="header"
  >
    <th
      className="ant-table-cell"
      colSpan={null}
      rowSpan={null}
      style={Object {}}
    >
      Actions
    </th>
  </Cell>
</tr>
`;

exports[`CreateEditProduct Page renders correctly: table rows 1 1`] = `
<tr
  className="ant-table-placeholder"
  style={
    Object {
      "display": null,
    }
  }
>
  <Cell
    colSpan={3}
    component="td"
    prefixCls="ant-table"
  >
    <td
      className="ant-table-cell"
      colSpan={3}
      rowSpan={null}
      style={Object {}}
    >
      <Empty
        image={<Simple />}
      >
        <LocaleReceiver
          componentName="Empty"
        >
          <div
            className="ant-empty ant-empty-normal"
          >
            <div
              className="ant-empty-image"
            >
              <Simple>
                <svg
                  className="ant-empty-img-simple"
                  height="41"
                  viewBox="0 0 64 41"
                  width="64"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fillRule="evenodd"
                    transform="translate(0 1)"
                  >
                    <ellipse
                      className="ant-empty-img-simple-ellipse"
                      cx="32"
                      cy="33"
                      rx="32"
                      ry="7"
                    />
                    <g
                      className="ant-empty-img-simple-g"
                      fillRule="nonzero"
                    >
                      <path
                        d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                      />
                      <path
                        className="ant-empty-img-simple-path"
                        d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                      />
                    </g>
                  </g>
                </svg>
              </Simple>
            </div>
            <p
              className="ant-empty-description"
            >
              No Data
            </p>
          </div>
        </LocaleReceiver>
      </Empty>
    </td>
  </Cell>
</tr>
`;
